### PR TITLE
feat: charge the user for the payment

### LINKED ISSUE #322: start charging user using charge API and order identifier
working on it 👍

### PR DESCRIPTION
**Pull request by nikit19**
fixes #322

### CODE REVIEW

**Comment by nikit19** [comment]
@iamareebjamal @dreadpool2 @simarsingh24 
PR updated according to the latest changes in the server.
Please review

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt [hunk 1]
```diff
@@ -151,16 +151,21 @@ class AttendeeViewModel(private val attendeeService: AttendeeService, private va
...
                 }.doFinally {
                     progress.value = false
                 }.subscribe({
-                    message.value = "Payment completed successfully!"
-                    Timber.d("Success placing order!")
+                    if (it.status != null && it.status) {
+                        message.value = it.message
+                        Timber.d("Successfully  charged for the order!")
+                    } else {
+                        message.value = it.message
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: This call is in both blocks

What's the use of if statement

Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Updated the PR but I am still using the if statements to see if the status is true/ false to see if the payment is successful  or not. Please review.

File: app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt [hunk 2]
```diff
@@ -147,6 +150,28 @@ class AttendeeViewModel(private val attendeeService: AttendeeService, private va
...
+                .doOnSubscribe {
+                    progress.value = true
+                }.doFinally {
+                    progress.value = false
+                }.subscribe({
+                    message.value = it.message
+                    if (it.status != null && it.status) {
+                        Timber.d("Successfully  charged for the order!")
+                    } else {
+                        Timber.d("Failed charging the user")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: So no feedback to user?

Comment (nikit19) [reply app/src/main/java/org/fossasia/openevent/general/attendees/AttendeeViewModel.kt]: Actually I am showing the feedback to the user. I am storing it in the message variable before the if else statements

**Comment by nikit19** [comment]
Resolved conflicts

**Pull request merged**